[Security Solution] Integrate new navigation in stateful

semd opened the pull request:
## Summary

issue: https://github.com/elastic/kibana/issues/179162

This PR brings the new navigation, with the solution-centric IA (Information Architecture), to the ESS (stateful) Security Solution.

To do so, the implementation of the `navigationTree$`, which was previously only implemented in serverless, has been integrated inside the generic "security_solution" plugin, so now it is available for ESS and serverless offerings.

In ESS users can still choose the navigation version, so we have to temporarily keep supporting both, the classic and new navigation implementations. After the rollout, the classic navigation components will be removed and the unified links architecture should be reassessed. The issue for the cleaning: https://github.com/elastic/kibana/issues/179572

### Rollout

The new solutions navigation will not be available for customers on 8.14, it will only be enabled for internal Elastic users (via Lauchdarkly), who will have the ability to opt out of it from their profile menu. We'll collect feedback and telemetry and address any bugs or improvements (together with the Kibana platform team). The plan is to start making it available to customers in 8.15.

### Testing

Unless we add the _kibana.yml_ configurations to enable the new navigation, the regular classic navigation will be displayed, it should keep working the same way without any change. The new landing pages (`Assets`, `Investigations`, `Machine Learning`,...), that exist only when using the new navigation, should not be accessible using the classic version.

To enable the new navigation add the following _kibana.yml_ configs:
```
xpack.cloud_integrations.experiments.enabled: true
xpack.cloud_integrations.experiments.flag_overrides:
  "navigation.solutionNavEnabled": true

xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```
And enable the advanced setting

![image](https://github.com/elastic/kibana/assets/17747913/07e8952d-5bd5-4700-8105-7732f08de28e)

### Screenshots

The app switcher:

<img width="293" alt="app switcher" src="https://github.com/elastic/kibana/assets/17747913/0a638b8f-fdc0-4d1a-b8d3-607e487215f4">

---

New Assets landing page:

<img width="549" alt="assets landing" src="https://github.com/elastic/kibana/assets/17747913/17bc8a94-02b4-4996-b9f5-8731ba81ac43">

---

For `Stack Management` we set the nav panel flyout and the cards landing page, this is temporary until a decision on how to show Stack Management links is made:

<img width="954" alt="stack management" src="https://github.com/elastic/kibana/assets/17747913/27ce6534-0508-4804-b224-8dc409042825">

---

The switch to go back to the classic nav is in the profile menu at the top-right corner:

![nav_switch](https://github.com/elastic/kibana/assets/17747913/f547b051-4924-42da-b12f-e308a4da5868)

semd committed:
unify navigation in security_solution plugin

semd committed:
fix some tests

semd committed:
Merge remote-tracking branch 'upstream/main' into 179162_security_stateful_solution_navigation

semd commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

semd committed:
fix dependencies

semd commented:
/ci

semd committed:
add stack management landing

semd committed:
fix translation keys

semd commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

semd committed:
translation fixes

semd commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

semd committed:
management landing only with solution nav

semd commented:
/ci

semd committed:
fix types

semd committed:
lazy load solution nav links

semd committed:
Merge remote-tracking branch 'upstream/main' into 179162_security_stateful_solution_navigation

semd commented:
/ci

semd committed:
fix jest test

semd committed:
reduce bundle size

semd commented:
/ci

semd committed:
fix mock type

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

semd commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 179162_security_stateful_solution_navigation

semd committed:
Merge remote-tracking branch 'upstream/main' into 179162_security_stateful_solution_navigation

sebelga reviewed (approved):
Changes to navigation types LGTM 👍

semd committed:
add stack management nav panel

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

semd commented on x-pack/packages/security-solution/navigation/src/constants.ts:
```diff
@@ -15,3 +15,88 @@ export enum LinkCategoryType {
+ * This is optional and only needed if the path is not registered in the plugin's `deepLinks`. e.g. `integrations:/browse/security`
+ * The path should not be used for links displayed in the main left navigation, since highlighting won't work.
+ **/
+export enum ExternalPageName {
```
This enum has been moved from _security_solution_serverless_ plugin to the security navigation package, so we can import and use it from any plugin.

semd commented on x-pack/plugins/security_solution/public/plugin_contract.ts:
```diff
@@ -21,66 +19,64 @@ export class PluginContract {
   public upsellingService: UpsellingService;
   public onboardingPageService: OnboardingPageService;
-  public extraRoutes$: BehaviorSubject<RouteProps[]>;
-  public appLinksSwitcher: AppLinksSwitcher;
-  public deepLinksFormatter?: DeepLinksFormatter;
```
Not needed anymore in the contract, now they are implemented inside the _security_solution_ plugin

semd commented on x-pack/plugins/security_solution/public/plugin_contract.ts:
```diff
@@ -21,66 +19,64 @@ export class PluginContract {
   public componentsService: ContractComponentsService;
   public upsellingService: UpsellingService;
   public onboardingPageService: OnboardingPageService;
-  public extraRoutes$: BehaviorSubject<RouteProps[]>;
```
We don't have serverless specific routes anymore, all routes are defined in the _security_solution_ plugin

semd commented on x-pack/plugins/security_solution/public/plugin_services.ts:
I've reorganized the services logic from _public/plugin.tsx_ into its own `PluginServices` class due to its increasing complexity. 
This change aims to separate concerns, leaving _plugin.tsx_ to handle registrations primarily. The result is code that's not only more readable but also easier to maintain.

semd commented on x-pack/plugins/security_solution/public/app/solution_navigation/categories.ts:
Everything inside `x-pack/plugins/security_solution/public/app/solution_navigation/` has been moved from `x-pack/plugins/security_solution_serverless/public/navigation/`

angorayc commented:
I found a minor unexpected behaviour of the header: The bubble was moved out of the viewport when scrolling:



https://github.com/elastic/kibana/assets/6295984/ae747a76-69a8-4e7c-8855-752cf3624f09

semd committed:
enable navigation version toggling

PhilippeOberti commented:
@semd I haven't looked at the code deeply yet, but glancing over it I could not see for sure if we're using telemetry to see who is using this new navigation. Is that something we would need to drive the decision on when to remove the old navigation code?

semd commented:
@PhilippeOberti 
> I could not see for sure if we're using telemetry to see who is using this new navigation. Is that something we would need to drive the decision on when to remove the old navigation code?

Yes! We'll need telemetry to make informed decisions in the future, but this is a cross-solution effort and the logic to toggle the `classic` / `new` navigation is owned by Kibana platform (shared-ux) team. They are going to take care of that, we don't have to do anything at the solution level. But good question anyway 👍

semd committed:
fix mock

semd committed:
Merge remote-tracking branch 'upstream/main' into 179162_security_stateful_solution_navigation

semd commented on x-pack/plugins/security_solution_ess/public/navigation/side_navigation.ts:
```diff
@@ -0,0 +1,112 @@
+        const footerGroup: GroupDefinition | undefined = draft.footer?.find(
+          ({ type }) => type === 'navGroup'
+        ) as GroupDefinition;
+        const management = footerGroup?.children.find((child) => child.link === 'management');
+        if (management) {
+          management.renderAs = 'panelOpener';
+          management.id = 'stack_management';
+          management.spaceBefore = null;
+          management.children = stackManagementLinks;
+        }
```
This injects the static "stack management" links in a hacky way, this code is temporary and its only purpose is to showcase the panel flyout for this section. 

Since the new navigation will be disabled for all customers on 8.14 (it will only be available to internal Elastic users), we can merge this code as is. Once a decision is made on how to show "stack management" links this code will be cleaned.

semd committed:
fix flickering when switching navigation version

semd commented on x-pack/plugins/security_solution/public/app_links.ts:
The main app_links file has been moved directly under _public/_ dir for consistency with the rest of the _plugins.tsx_ lazy-loaded resources.

semd commented on x-pack/plugins/security_solution/public/common/components/security_route_page_wrapper/index.tsx:
```diff
@@ -6,15 +6,17 @@
 
 interface SecurityRoutePageWrapperProps {
   pageName: SecurityPageName;
+  redirectOnMissing?: boolean;
```
When switching the navigation version users may be located on a page that does not exist. In this scenario, we should redirect the user to the home page (get started).

sebelga replied:
Not sure how your logic works but just want to point out that, in case a page does not exist in the current solution tree definition, we look for a different solution where the page _might_ exist. If we find it we redirect to that solution. 
Maybe that does not apply in this context but wanted to share it.

semd commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 179162_security_stateful_solution_navigation

Commit:
[ObsAiAssistant] Add “Manage connectors” button to flyout (#179834)

Follow-up to https://github.com/elastic/kibana/pull/179380

This adds a link "Manage connectors" to the flyout where users can
select a connector.


![image](https://github.com/elastic/kibana/assets/209966/8016b1d2-84d7-4372-909e-636677c0de4d)

vitaliidm committed:
[Security Solution][Detection Engine] fixes risk score enrichment for rules with suppression (#180359)

## Summary

Fixes risk score host enrichment for query, threshold, IM rules when
suppression is enabled


Setting Feature Flag `isNewRiskScoreModuleAvailable` to [true by
default](https://elastic.slack.com/archives/C056TQ5J81Y/p1694699742267509)
did not work since enrichment
[function](https://github.com/elastic/kibana/blob/8.13/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/enrichments/index.ts#L41)
checks for a flag and if experimentalFeatures property undefined, it
sets `isNewRiskScoreModuleAvailable` to `false`.

Also adds test coverage for new risk score module and for asset
criticality as well

### Flaky tests runner

[FTR ESS
x200](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/5634)
[FTR Serverless x200

](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/5635)

### Checklist


- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

jennypavlova committed:
[Infra] Implement dashboard tab UI in host details (#178518)

Closes #176070
Closes #178319
Closes [#175447](https://github.com/elastic/kibana/issues/175447)

## Summary 

This PR adds a dashboard tab to the asset details view. This is the
first version of the tab and it looks similar to the APM solution in
service overview.

⚠️ After [this PR](https://github.com/elastic/kibana/pull/179576) was
merged the structure of the API changed and the saved object is now one
per linked dashboard (not one per asset type as before). Those changes
will give us more flexibility in the future and the endpoints allow us
to edit/link/unlink a dashboard easier than before (and we don't have to
get and iterate through all dashboards when updating/deleting)

The new structure of the saved object is now : 
```javascript
    "properties": {
       "assetType": {
         "type": "keyword"
       },
       "dashboardSavedObjectId": {
         "type": "keyword"
       },
       "dashboardFilterAssetIdEnabled": {
         "type": "boolean"
       }
     }
```

This initial implementation will show the dashboard tab **ONLY** if the
feature flag (`enableInfrastructureAssetCustomDashboards`) is enabled
(this will change)

## Updates: 
- #178319 New splash screen

<img width="1909" alt="image"
src="https://github.com/elastic/kibana/assets/14139027/e595c4bb-fdbb-415b-b778-3dc39af20b54">

- The new API endpoints added in [this
PR](https://github.com/elastic/kibana/pull/179576) are now used
- Fix switching dashboards not refreshing the content and not applying
filters issue

## Next steps 

- [ ] [[Infra] Dashboard locator |
kibana#178520](https://github.com/elastic/kibana/issues/178520)
- [ ] [[Infra] Dashboard feature activation |
kibana#175542](https://github.com/elastic/kibana/issues/175542)


## Testing 
- Generate some hosts with metrics: 
- `node scripts/synthtrace --clean --scenarioOpts.numServices=5
infra_hosts_with_apm_hosts.ts`
   - or use metricbeat/remote cluster

- Enable the `enableInfrastructureAssetCustomDashboards` feature flag

- Go to Hosts view flyout / Go to Asset details page
   - Link a dashboard
   - Edit the linked dashboard (enable/disable filter by hostname)
   

https://github.com/elastic/kibana/assets/14139027/ad6b87aa-e2de-42fa-9565-4bfe32ffd146

   - Unlink a dashboard: In case of unlinking: 
       - single dashboard -> empty state 
       - multiple dashboards -> other dashboard


https://github.com/elastic/kibana/assets/14139027/4f39f3aa-b7fa-407d-8991-79d19d3ee076

- Navigation between Hosts view flyout / Asset details page (Click `Open
as page`) and persisting the state:


https://github.com/elastic/kibana/assets/14139027/98756cb0-7675-4bc0-9e14-0fb8d95cce30

- Link custom dashboard (create a dashboard and link it after)

<img width="1914" alt="image"
src="https://github.com/elastic/kibana/assets/14139027/9739f355-39b9-4b1d-b98e-0a47db9095df">

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kpatticha committed:
[APM] Format comparison value for all overview mobile metrics (#180353)

## Summary

Fixes https://github.com/elastic/kibana/issues/174551

### before 
![Screenshot 2024-04-09 at 12 10
11](https://github.com/elastic/kibana/assets/3369346/4ebc02f0-f2ff-43e6-877b-27756272a5ab)

### after 
![Screenshot 2024-04-09 at 12 23
49](https://github.com/elastic/kibana/assets/3369346/990f08da-8e7a-40db-9fdd-708c19fa2c3a)

Commit:
[Serverless] Remove visualisation link from side navs (#180387)

kc13greiner committed:
Upgrading express from 4.17.3 --> 4.19.2 (#180373)

## Summary

Upgrading `express` from 4.17.3. to 4.19.2


[Changelog](https://github.com/expressjs/express/compare/4.17.3...4.19.2)

Commit:
[Stateful sidenav] Put new nav behind launch darkly flag (#179913)

justinkambic committed:
[Observability Onboading] Export additional types for usage with Package List Grid (#180416)

## Summary

Related to https://github.com/elastic/kibana/pull/179573, exporting
these additional types make downstream usage of the `PackageListGrid`
component and `useAvailablePackages` hook easier.

Co-authored-by: Julia Bardi <90178898+juliaElastic@users.noreply.github.com>

paul-tavares committed:
[Security Solution][Endpoint] Fix Sentinelone `isolate`/`release` new action completion logic so that its gated by feature flag (#180390)

## Summary

- Updates both `release` and `isolate` actions for SentinelOne so that
the new (`8.14.0`) approach for completing the response actions is gated
by the `responseActionsSentinelOneV2Enabled` feature flag

nreese committed:
[embeddable rebuild] useBatchedOptionalPublishingSubjects (#180221)

Fixes https://github.com/elastic/kibana/issues/180088

1. Changes observable pipe from `debounceTime(), skip(1)` to `skip(1),
debounceTime()`. Updates test to verify this change results in
subscription getting fired when observable.next is called before
debounceTime fires.
2. rename `useBatchedPublishingSubjects` to
`useBatchedOptionalPublishingSubjects`. Remove `useMemo` since spreading
subjects results in new array every time and useMemo does nothing.
3. Update `PresentationPanelInternal` to use
`useBatchedOptionalPublishingSubjects`
4. create new `useBatchedPublishingSubjects` that types subjects as
`PublishingSubject[]`. New

---------

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dmlemeshko committed:
[FTR] add dataViews service (#180066)

## Summary

This PR makes few changes:
- extend
[kbn-test-subj-selector](https://github.com/elastic/kibana/compare/main...dmlemeshko:kibana:ftr/add-dataViews-service?expand=1#diff-43f2401dd3f9c11b6cbd75c8801a6ccbbe1d4db5a19e907f263c4932f810c73c)
with wildcard (*) support:

```
testSubjSelector('*dataView-switch-link') => [data-test-subj*="dataView-switch-link"]
```

It allows us to search for DOM elements with the common text part in
`data-test-subj` attribute, e.g. `lns-dataView-switch-link` &
`discover-dataView-switch-link`

- add new FTR service
[dataViews](test/functional/apps/discover/group4/_adhoc_data_views.ts)
to unify mostly identical code related to data views across multiple
page objects (lens, discover, unified_search, dashboard)
It is not a big win in terms of code cleanup, but should save some time
whenever we will need to make logic updates.

juliaElastic committed:
[Fleet] added Fleet synthetic check to staging quality gates (#180461)

## Summary

Closes https://github.com/elastic/ingest-dev/issues/3065

Added Fleet synthetic monitor check to Kibana staging quality gates.
It has been stable in the past two weeks, added with soft fail for now.

This monitor verifies that a long running project is healthy in staging.
It aims to flag issues if there is a breaking change in Kibana / Fleet
plugin.

marshallmain committed:
[Security Solution] Classify EQL verification and ML job missing errors as user errors (#180094)

## Summary

Building on https://github.com/elastic/kibana/pull/180040

Detection rules commonly fail when prebuilt rules are imported and
enabled without the appropriate indices (for EQL) or ML jobs (for ML
rules). EQL rules fail with a `verification_exception` because the EQL
search API validates the fields in the search request against the
indices in the request; if there are no indices then it returns an
exception. ML rules fail with a `<job name> missing` exception on the
search request if the job is not found. Both of these errors do not mean
that the system is overloaded or performing incorrectly in some way, but
they are still showing up in large volumes on SLO dashboards.

This PR builds on #180040, which introduces the ability to classify
errors as "user errors" when the error is not due to some kind of system
malfunction, but more related to incorrect (or insufficient) user
actions.

### Testing
#### EQL
1. Create 2 indices, `test` and `test2`
```
PUT /test
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      }
    }
  }
}

PUT /test2
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      },
      "event.category": {
        "type": "keyword"
      }
    }
  }
}
```

2. Create (disabled) an EQL rule that queries `test*` and uses a simple
query like `file where true`
3. Delete the index `test2`
4. Enable the rule. The rule will fail with a `verification_exception`
because `test` does not have `event.category`.
5. Use your favorite debugging method to verify that `userError` was
`true` in `addLastRunError` in
`x-pack/plugins/alerting/server/monitoring/rule_result_service.ts`
(hopefully this property will be added to the rule SO so we can check it
there in API integration tests)

#### ML rules
1. Import a prebuilt ML rule (`Unusual Process Spawned by a User`, for
example)
2. Enable the rule. The rule will fail with `An error occurred during
rule execution: message: "problem_child_rare_process_by_user missing"`
3. Use your favorite debugging method to verify that `userError` was
`true` in `addLastRunError` in
`x-pack/plugins/alerting/server/monitoring/rule_result_service.ts`
(hopefully this property will be added to the rule SO so we can check it
there in API integration tests)

miltonhultgren committed:
[Obs AI Assistant] Support user instructions via Knowledge base or API request (#180263)

Part of #179736

### Summary

This PR adds the initial support for user instructions by:

1. Reading entries from the Knowledge base marked with label `category`
having the value `instruction`
2. Accepting instructions via the `/chat/complete` endpoint
3. Merging these two sources of instructions, giving priority to those
passed via the request to `/chat/complete`

### How to test
Using either the UI or the API, make a request to `/chat/complete`
asking about any topic.

Then create a basic instruction inside the Knowledge base that should
alter the LLMs interpretation of the question:
```
POST kbn:/internal/observability_ai_assistant/kb/entries/save
{
  "id": "my_instruction",
  "text": "While your main role is to help with Observability tasks, feel free to share your knowledge about Lord of the rings with the user.",
  "labels": {
   "category": "instruction" 
  }
}
```

Then make another request to `/chat/complete` and see how the response
changed.

To validate that instructions sent via the request take priority, make
another request that specifies the same `doc_id` as the one entered into
the knowledge base and repeat the request.

semd committed:
address comment

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

semd committed:
Revert "[Obs AI Assistant] Support user instructions via Knowledge base or API request (#180263)"

This reverts commit ab2132cdca2383a5c81255dcdd91c94af2deaa68.

semd committed:
Revert "[Security Solution] Classify EQL verification and ML job missing errors as user errors (#180094)"

This reverts commit 9242afb7a837c16b0a2ad11777246a2ee7c7ae59.

semd committed:
Revert "[Fleet] added Fleet synthetic check to staging quality gates (#180461)"

This reverts commit 8ef7debeb4fa50562c6d70b719d7f0b26cd6afa8.

semd committed:
Revert "[FTR] add dataViews service (#180066)"

This reverts commit 354b76d42863032c0cb140c41f2d4eb9f15b226b.

semd committed:
Revert "[embeddable rebuild] useBatchedOptionalPublishingSubjects (#180221)"

This reverts commit 51c3fa5c55f8a838ce86feb1b1dfbea8383315b5.

semd committed:
Revert "[Security Solution][Endpoint] Fix Sentinelone `isolate`/`release` new action completion logic so that its gated by feature flag (#180390)"

This reverts commit 15942cded32cee51084fecb14c4b4ef08a2fef10.

semd committed:
Revert "[Observability Onboading] Export additional types for usage with Package List Grid (#180416)"

This reverts commit e003f1ae9a316696f88e405ddcd3a0431a64f8f1.

semd committed:
Revert "[Stateful sidenav] Put new nav behind launch darkly flag (#179913)"

This reverts commit e07a7d1fd6f5768fac2d88865d379004a9a0b8e6.

semd committed:
Revert "Upgrading express from 4.17.3 --> 4.19.2 (#180373)"

This reverts commit fcbfbaa080b92316745a38386db34943be3cb51f.

semd committed:
Revert "[Serverless] Remove visualisation link from side navs (#180387)"

This reverts commit 9fd28f62bb40dec618c16b1f51dc0b3983afcac8.

semd committed:
Revert "[APM] Format comparison value for all overview mobile metrics (#180353)"

This reverts commit c376a6c6de77e88995320c3c0e1d71a81b72aacd.

semd committed:
Revert "[Infra] Implement dashboard tab UI in host details (#178518)"

This reverts commit d48a45dbf14ccd3fbe770fed847f412f0e8bb474.

semd committed:
Revert "[Security Solution][Detection Engine] fixes risk score enrichment for rules with suppression (#180359)"

This reverts commit 20d53854d92e44ec0d9ea617570841cc4e3d4baa.

semd committed:
Revert "[ObsAiAssistant] Add “Manage connectors” button to flyout (#179834)"

This reverts commit 18fbcb2f1af4979d1fef15ee13d3567a44721f34.

semd commented:
sorry for the noise

semd committed:
conflict solved

angorayc commented:
I wasn't able to find the expected page from the search box: `File data visualizer` and `Data view data visualizer`, but if I put `Data Visualizer` and `ES|QL Data Visualizer`, the results were correct.

semd committed:
fix registring security navigation tree

semd commented:
@angorayc 
> I wasn't able to find the expected page from the search box: `File data visualizer` and `Data view data visualizer`, but if I put `Data Visualizer` and `ES|QL Data Visualizer`, the results were correct.

The global search result titles of those links depend on the way the ML plugin registers their deep links. On the other hand, the titles in the Security Solution navigation & breadcrumbs for ML links depend on the `navigationTree` config we define on our side. 

The titles of those links were set when we implemented the ML landing page in serverless some time ago. So I think that if we want to address those inconsistencies it should be done it in a separate issue, for both serverless and ess navigations.

semd commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 179162_security_stateful_solution_navigation

angorayc commented on x-pack/plugins/security_solution/public/assets/routes.tsx:
```diff
@@ -0,0 +1,30 @@
+const AssetsPage = React.memo(function AssetsPage() {
+  return (
+    <PluginTemplateWrapper>
+      <SecurityRoutePageWrapper pageName={SecurityPageName.assets} redirectOnMissing>
```
👍

angorayc commented on x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_security_solution_navigation.test.tsx:
```diff
@@ -29,23 +34,57 @@ describe('Security Solution Navigation', () => {
         canBeCollapsed: true,
         name: 'Security',
         icon: 'logoSecurity',
+        children: expect.anything(),
```
I'd recommend to mock  `SecuritySideNav` from '../security_side_nav' to expect an explicit value here

semd replied:
I've tried to do this, the `SecuritySideNav` component is already mocked, but `children` is a _ReactNode_, and `isEqual` crashes when trying to assert it. So I decided to check only the regular props here, making sure the `children` exists, and then I added an assertion for the `children`, starting at line 63:

```
      // check rendering of SecuritySideNav children
      const { findByTestId } = render(<>{result.current?.children}</>);

      expect(mockSecuritySideNav).toHaveBeenCalled();
      expect(await findByTestId('SecuritySideNav')).toBeInTheDocument();
```

If you know any better way to do this please let me know.

angorayc reviewed (approved):
Tested locally, LGTM! Thanks for implementing the new side navigation @semd !

ashokaditya commented on x-pack/plugins/security_solution/public/common/lib/kibana/services.ts:
```diff
@@ -11,9 +11,24 @@ import type { StartPlugins } from '../../../types';
 type GlobalServices = Pick<CoreStart, 'application' | 'http' | 'uiSettings' | 'notifications'> &
   Pick<StartPlugins, 'data' | 'unifiedSearch' | 'expressions' | 'savedSearch'>;
 
+/**
+ * This class is a singleton that holds references to core Kibana services.
+ * It is initialized during the plugin start lifecycle.
+ * Use with caution since it is not updated if services are changed after initialization.
+ * useKibana hook should be used in React components to access these services.
+ */
 export class KibanaServices {
+  /**
+   * Whether the environment is 'serverless' or 'traditional'
+   */
   private static buildFlavor?: string;
+  /**
+   * The current Kibana branch. e.g. 'main'
+   */
   private static kibanaBranch?: string;
+  /**
+   * The current Kibana version. e.g. '8.0.0' or '8.0.0-SNAPSHOT'
+   */
   private static kibanaVersion?: string;
   private static prebuiltRulesPackageVersion?: string;
   private static services?: GlobalServices;
```
Thanks for adding comments. :fire:

ashokaditya reviewed (approved):
Thanks for doing this. I tested out the changes with the kibana config and it works as expected. 

_Some early feedback, I had a bit of confusion finding management pages using the new nav (as it is tucked under Management link at the bottom left nav item), then I found it under Assets (looking through your screenshots). And I think it makes sense to group Fleet, Endpoint, Cloud under Assets._ 🥇

semd commented:
@ashokaditya 
> _Some early feedback, I had a bit of confusion finding management pages using the new nav (as it is tucked under Management link at the bottom left nav item), then I found it under Assets (looking through your screenshots). And I think it makes sense to group Fleet, Endpoint, Cloud under Assets._ 🥇

Yes, it's not the first time we have received this feedback, we are aware of that. We will probably implement a guided tour for the new nav, showing where all the links are.

Thanks for taking the time to test it.
